### PR TITLE
Removing independent save buttons on engagement form

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
Issue #: https://apps.itsm.gov.bc.ca/jira/browse/DESENG-507

*Description of changes:*
    - The existing "Save" button in the floating bar has been split into two distinct actions: "Save and Continue" and "Save and Exit".
    - Tabs are greyed out, and widgets are disabled until the engagement is successfully saved. A helpful tool-tip has been added to inform users that the engagement needs to be saved before accessing certain features.
    - Independent save buttons previously present in tabs, such as "Additional Details", "Settings" have been removed. Now, the floating save bar is universally employed when editing an engagement.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/engagement/form/EngagementFormTabs/EngagementTabsContext.tsx [hunk 1]
```diff
@@ -240,13 +280,15 @@ export const EngagementTabsContextProvider = ({ children }: { children: React.Re
             ) as PatchEngagementSettingsRequest;
 
             if (Object.keys(updatedSettings).length === 0) {
+                setSaving(false);
                 return Promise.resolve();
```
Comment (Baelx) [inline met-web/src/components/engagement/form/EngagementFormTabs/EngagementTabsContext.tsx:284]: I don't think we need to call `Promise.resolve()` here. Within an `async/await` function, a promise will always be returned automatically.

```suggestion
                return;
```

File: met-web/src/components/engagement/form/EngagementFormTabs/EngagementTabsContext.tsx [hunk 2]
```diff
@@ -276,6 +321,39 @@ export const EngagementTabsContextProvider = ({ children }: { children: React.Re
...
+            const updatedSlug = updatedDiff(
+                {
+                    slug: savedSlug,
+                },
+                slug,
+            ) as EngagementSlugPatchRequest;
+
+            if (Object.keys(updatedSlug).length === 0) {
+                setSaving(false);
+                return Promise.resolve();
```
Comment (Baelx) [inline met-web/src/components/engagement/form/EngagementFormTabs/EngagementTabsContext.tsx:336]:
```suggestion
                return;
```

File: met-web/src/components/engagement/form/EngagementFormTabs/EngagementTabsContext.tsx [hunk 3]
```diff
@@ -276,6 +321,39 @@ export const EngagementTabsContextProvider = ({ children }: { children: React.Re
...
+            }
+
+            const response = await patchEngagementSlug({
+                ...updatedSlug,
+                engagement_id: savedEngagement.id,
+            });
+            setSavedSlug(response.slug);
+            setSaving(false);
+            return Promise.resolve();
```
Comment (Baelx) [inline met-web/src/components/engagement/form/EngagementFormTabs/EngagementTabsContext.tsx:345]:
```suggestion
            return;
```

File: met-web/src/components/engagement/form/EngagementFormTabs/Settings/InternalEngagement.tsx [hunk 4]
```diff
@@ -1,15 +1,25 @@
...
-        setIsInternal(e.target.checked);
+    useEffect(() => {
+        setInitialInternalFlag(engagementFormData.is_internal);
+    }, []);
+
+    const handleChangeIsInternal = () => {
+        setInitialInternalFlag(!initialInternalFlag);
+        setEngagementFormData({
+            ...engagementFormData,
+            is_internal: !initialInternalFlag,
```
Comment (Baelx) [inline met-web/src/components/engagement/form/EngagementFormTabs/Settings/InternalEngagement.tsx:21]:
```suggestion
            isInternal: !initialInternalFlag,
```

File: met-web/src/components/engagement/form/EngagementFormTabs/EngagementTabsContext.tsx [hunk 5]
```diff
@@ -41,6 +47,14 @@ const initialEngagementFormData = {
     consent_message: '',
 };
 
+const initialEngagementSettingsFormData = {
+    send_report: false,
```
Comment (Baelx) [inline met-web/src/components/engagement/form/EngagementFormTabs/EngagementTabsContext.tsx:51]:
```suggestion
    sendReport: false,
```

File: met-web/src/components/engagement/form/EngagementFormTabs/Settings/SendReport.tsx [hunk 6]
```diff
@@ -1,22 +1,26 @@
...
 import { useAppDispatch } from 'hooks';
 import { openNotification } from 'services/notificationService/notificationSlice';
-import { EngagementSettingsContext } from './EngagementSettingsContext';
 
 const SendReport = () => {
-    const { settings, settingsLoading } = useContext(EngagementTabsContext);
+    const { settings, sendReport, setSendReport, settingsLoading } = useContext(EngagementTabsContext);
     const { savedEngagement } = useContext(ActionContext);
-    const { sendReport, setSendReport } = useContext(EngagementSettingsContext);
+    const [initialSendReportFlag, setInitialSendReportFlag] = useState(settings.send_report);
```
Comment (Baelx) [inline met-web/src/components/engagement/form/EngagementFormTabs/Settings/SendReport.tsx:12]: Is there a reason why we're using two different syntaxes for sendReport? i.e. `sendReport` and `send_report`?

**Review by Baelx — APPROVED** [review]
Great work! 

Consider removing the `Promise.resolve()` calls where indicated (unless I'm wrong about them being necessary), and correcting the snake case syntax.

File: met-web/src/components/engagement/form/EngagementFormTabs/Settings/SendReport.tsx (hunk 6 above)
Comment (VineetBala-AOT) [reply met-web/src/components/engagement/form/EngagementFormTabs/Settings/SendReport.tsx:12]: Yes, it is related to the property coming from the DB. Similar is the case with is_internal.

**Pull request merged**